benchexec/tablegenerator/react-table/src/utils/utils.js

```diff
@@ -192,6 +192,355 @@ const setHashSearch = (
+  // with
+  // <idFilter> := id(values(<value>+))
+  // <runsetFilter> := <runsetId>(<columnFilter>+)[,]
+  // <columnFilter> := <columnId>*<name>*(<filter>)[,]
+  // <filter> := <valueFilter>|<statusColumnFilter>
+  // <valueFilter> := value(<value>)
+  // <statusColumnFilter> := <statusFilter>|<categoryFilter>|<statusFilter>,<categoryFilter>
+  // <statusFilter> := status(in(<value>+)|notIn(<value>+)|empty())
+  // <categoryFilter> := category(in(<value>+)|notIn(<value>+)|empty())
+  // <value> := <urlencodedTerminalValue>
+
+  // one transformed example would be
+  // 1(0*status*(statusFilter(notIn(true)),categoryFilter(in(correct,missing))),1*cputime*(value(%3A1120)))
+
+  const { ids, ...rest } = groupedFilters;
+  const runsetFilters = [];
+  if (ids) {
+    runsetFilters.push(`id(values(${ids.values.map(escape).join(",")}))`);
+  }
+  for (const [tool, column] of Object.entries(rest)) {
+    const columnFilters = [];
+    for (const [columnId, filters] of Object.entries(column)) {
+      const columnFilterHeader = `${columnId}*${filters.name}*`;
+      let filter;
+      if (filters.statusValues || filters.categoryValues) {
+        // <statusColumnFilter>
+        const statusColumnFilter = [];
+        const { statusValues, categoryValues } = filters;
+        const toolStatusValues = allStatusValues[tool][columnId];
+        const toolCategoryValues = allCategoryValues[tool][columnId];
+
+        const hasStatusFilter = !!statusValues;
+        const hasStatusUnchecked =
+          hasStatusFilter && statusValues.length !== toolStatusValues.length;
+
+        const hasCategoryFilter = !!categoryValues;
+        const hasCategoryUnchecked =
+          hasCategoryFilter &&
+          categoryValues.length !== toolCategoryValues.length;
+
+        if (hasStatusFilter) {
+          if (hasStatusUnchecked) {
+            const encodedFilter = createDistinctValueFilters(
+              statusValues,
+              toolStatusValues,
+            );
+            statusColumnFilter.push(`status(${encodedFilter})`);
+          }
+          if (!hasCategoryFilter) {
+            statusColumnFilter.push("category(empty())");
+          }
+        }
+        if (hasCategoryFilter) {
+          if (!hasStatusFilter) {
+            statusColumnFilter.push("status(empty())");
+          }
+          if (hasCategoryUnchecked) {
+            const encodedFilter = createDistinctValueFilters(
+              categoryValues,
+              toolCategoryValues,
+              true,
+            );
+            statusColumnFilter.push(`category(${encodedFilter})`);
+          }
+        }
+        filter = statusColumnFilter.join(",");
+      } else {
+        // <valueFilter>
+        filter = `value(${escape(filters.value)})`;
+      }
+      if (filter !== "") {
+        columnFilters.push(`${columnFilterHeader}(${filter})`);
+      }
+    }
+    if (columnFilters.length > 0) {
+      runsetFilters.push(`${tool}(${columnFilters.join(",")})`);
+    }
+  }
```

Comment:
Is there maybe a way to implement this part a little bit more compact? The large if-else tree makes this a little bit hard to read and possibly harder to maintain in the future. From just looking through it, I couldn't think of a better way immediately, but since you've implemented the URL filters yourself, maybe you can think of something. If not, it's also fine though.

Reply:
Hmmm... i also don't see an immediate way of making this logic more compact.
I extracted the `<statusColumnFilter>` into a separate function to make the overall control-flow more obvious. 
We could also try to flatten the tree a bit but i'm not sure if that would result in more readable code. 
e.g. `if(hasStatusFilter&&hasStatusUnchecked){...}; if(hasStatusFilter&&!hasCategoryFilter){...}` 

The other if statements like `if (filter !== "")` exist to reduce the length of the resulting serialized filter so i wouldn't want to remove them. 

If you think the code might be really hard to understand for future contributors though then i will 
try to rework this section of course.

Reply:
I agree with both of you :-)

Luckily, this part of the code is separate from the rest and could also be refactored later if we find a good way, there is no need to delay merging due to this.